cmd/event-received/sirius_event_handler_test.go

```diff
@@ -1926,3 +1926,155 @@ func TestHandleMaterialChangeConfirmed(t *testing.T) {
 		})
 	}
 }
+
+func TestHandleCertificateProviderIdentityCheckFailed(t *testing.T) {
+	event := &events.CloudWatchEvent{
+		DetailType: "certificate-provider-identity-check-failed",
+		Detail:     json.RawMessage(`{"uid":"M-1111-2222-3333"}`),
+	}
+
+	provided := &donordata.Provided{
+		PK:                  dynamo.LpaKey("123"),
+		SK:                  dynamo.LpaOwnerKey(dynamo.DonorKey("456")),
+		Type:                lpadata.LpaTypePropertyAndAffairs,
+		Donor:               donordata.Donor{ContactLanguagePreference: localize.En},
+		CertificateProvider: donordata.CertificateProvider{FirstNames: "a", LastName: "b"},
+		Tasks:               donordata.Tasks{SignTheLpa: task.StateCompleted},
+	}
+
+	dynamoClient := newMockDynamodbClient(t)
+	dynamoClient.EXPECT().
+		OneByUID(ctx, "M-1111-2222-3333", mock.Anything).
+		Return(nil).
+		SetData(dynamo.Keys{PK: dynamo.LpaKey("123"), SK: dynamo.DonorKey("456")})
+	dynamoClient.EXPECT().
+		One(ctx, dynamo.LpaKey("123"), dynamo.DonorKey("456"), mock.Anything).
+		Return(nil).
+		SetData(provided)
+
+	notifyClient := newMockNotifyClient(t)
+	notifyClient.EXPECT().
+		SendActorEmail(ctx, notify.ToDonor(provided), "M-1111-2222-3333", notify.InformDonorPaperCertificateProviderIdentityCheckFailed{
+			CertificateProviderFullName: "a b",
+			LpaType:                     "property and affairs",
+			DonorStartPageURL:           "app:///start",
+		}).
+		Return(nil)
+
+	bundle, err := localize.NewBundle("../../lang/en.json", "../../lang/cy.json")
```

Comment:
This was the only way I could think of testing this without being able to make a mock for Localizer (I tried various combos of interfaces but couldn't get a valid implementation of the type with our current setup)

Reply:
it is a bit of a pain, maybe we should make `localizer.Localizer` an interface? Then you'd easily be able to mock it